src/test/build-all-projects.ts

```diff
@@ -13,78 +16,92 @@ const ts = '--ts';
 const javascript = 'javascript';
 const typescript = 'typescript';
 const space = ' ';
-let jsTemplates = getDirectories('src/app/templates/js');
-let tsTemplates = getDirectories('src/app/templates/ts');
-let hostsTemplates = getDirectories('src/app/templates/hosts');
+let parsedProjectJsonData = jsonData.getParsedProjectJsonData();
+let projectTemplates = jsonData.getProjectTemplateNames();
+let hostsTemplates = jsonData.getHostTemplateNames();
 
 describe('Setup test environment for Yo Office build tests', () => {
-    it ('Install Yeoman Generator and Install local install of Yo Office and link', function(done){
+    it ('Install Yeoman Generator, install local instance of Yo Office and link', function(done){
         _setupTestEnvironment();
         done();
     });
-}); 
+});
+
+describe('Install and build projects', () => {        
+});
 
 // Build Typescript project types for all supported hosts
 for (var i = 0; i < hostsTemplates.length; i++)
 {
-    for (var j = 0; j < tsTemplates.length; j++)
+    for (var j = 0; j < projectTemplates.length; j++)
     {
-        describe('Install and build projects', () => {
-            let projectType = tsTemplates[j];        
-            let host = hostsTemplates[i];
-            let scriptType = typescript;
-            let projectName = projectType + host + typescript;
-            let projectFolder = path.join(__dirname, '/', projectName);          
-        
-            describe(stringBuildStart +  host + space + projectType + space + typescript, () => {
-                it(stringBuildSucceeds,function(done){  
-                    _generateProject(projectType, projectName, host, projectFolder, scriptType);
-                    _buildProject(projectFolder, projectType);
+        let host = hostsTemplates[i].toLowerCase();
```

Comment:
>let host = hostsTemplates[i].toLowerCase(); [](start = 8, length = 43)

nit: This can be executed in the **for** above, so this operation is not executed during the iterations of the project templates.

Reply:
fixed locally